httpclient/response.go

```diff
@@ -71,3 +84,92 @@ func WithResponseUnmarshal(response any) DoOption {
 		},
 	}
 }
+
+func findContentsField(response any, body *common.ResponseWrapper) (*reflect.Value, bool) {
+	value := reflect.ValueOf(response)
+	value = reflect.Indirect(value)
+	if value.Kind() != reflect.Struct {
+		return nil, false
+	}
+
+	// Check if there is a "Contents" of typo io.ReadCloser
+	// This is by internal convention with the teams.
+	ioType := reflect.TypeOf((*io.ReadCloser)(nil)).Elem()
+	contentField := value.FieldByName("Contents")
+	if !contentField.IsValid() || !contentField.CanSet() || contentField.Type() != ioType {
+		return nil, false
+	}
+	return &contentField, true
+}
+
+func injectHeaders(response any, body *common.ResponseWrapper) error {
+	value := reflect.ValueOf(response)
+	value = reflect.Indirect(value)
+	objectType := value.Type()
+
+	if objectType.Kind() != reflect.Struct {
+		return nil
+	}
+	headers := body.Response.Header
```

Comment:
Header is not a pointer field, so it cannot be nil.

```
type Response struct {

	Header Header. <-- Cannot be nil

        Request *Request <-- This could be nil

```